tests/func/test_stage.py

```diff
@@ -227,6 +227,13 @@ def test_parent_repo_collect_stages(tmp_dir, scm, dvc):
     assert deep_subrepo_stages != []
 
 
+def test_collect_repo_ignored_dir_unignored_pattern(tmp_dir, dvc, scm):
+    tmp_dir.gen({".gitignore": "data/**\n!data/**/\n!data/**/*.dvc"})
+    scm.add([".gitignore"])
+    (stage,) = tmp_dir.dvc_gen({"data/raw/tracked.csv": "5,6,7,8"})
+    assert dvc.stage.collect_repo() == [stage]
```

Comment:
This is fine for ensuring that `collect_repo()` works properly, but can you add a similar test to `tests/func/test_scm.py` that ensures the `!` exclusions work properly?

using this example the checks would be something like:
```
assert scm.is_ignored("data/")
assert scm.is_ignored("data/raw/")
assert not scm.is_ignored("data/raw/tracked.csv.dvc")
```

Reply:
Right, makes sense. Added one